iceoryx_examples/waitset_in_c/ice_c_waitset_gateway.c

```diff
@@ -66,17 +76,18 @@ int main()
     shutdownTrigger = iox_user_trigger_init(&shutdownTriggerStorage);
 
     // attach shutdownTrigger with no callback to handle CTRL+C
-    iox_ws_attach_user_trigger_event(waitSet, shutdownTrigger, 0U, NULL);
+    iox_ws_attach_user_trigger_event(waitSet, shutdownTrigger, 0U, shutdownCallback);
 
     //// register signal after shutdownTrigger since we are using it in the handler
     signal(SIGINT, sigHandler);
     signal(SIGTERM, sigHandler);
 
+    uint64_t sumOfAllSamples = 0U;
```

Comment:
Same atomic problem, but cannot use std::atomic. Would be something like `atomic_int` from `<stdatomic.h>` like in https://stackoverflow.com/questions/25319825/how-to-use-atomic-variables-in-c